ts/parts/LogarithmicAxis.ts

```diff
@@ -215,7 +215,7 @@ class LogarithmicAxis {
      */
     public static compose(AxisClass: typeof Axis): void {
 
-        /* eslint-disable no-invalid-this */
+        AxisClass.keepProps.push('logarithmic');
```

Comment:
I think this one actually makes sense 🤔I see there's code to be executed when switching from `log` to `linear` type. Nice catch! 👍

Reply:
I disagree with you. Saving constructor calls is sensible, too.

Reply:
But the code is clearing props only on `axis.linear` object. I'm not so convinced anymore.. :)

Reply:
Above comment was added before your loaded for me, I'm commenting myself comment above. Saying that, I don't understand what do you mean.

Reply:
I removed the destroy over design. :)
I mean this part here: https://github.com/highcharts/highcharts/blob/5133f3e50d90fac52261571ac43fd0c76a076de5/ts/parts/LogarithmicAxis.ts#L242-L244